File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.java
import java.util.HashMap;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        String[] answer = {};
        HashMap<String , Integer> playersMap = new HashMap<String, Integer>();
		
		for(int i = 0; i < players.length; i++) {
			playersMap.put(players[i], i);
		}
		
		for(int i = 0; i < callings.length; i++) {
			int playerNum = playersMap.get(callings[i]);
			
			if(playerNum != 0) {
				playersMap.put(players[playerNum], playerNum - 1);
				playersMap.put(players[playerNum - 1], playerNum);
				
				String tmp = players[playerNum];
				players[playerNum] = players[playerNum - 1];
				players[playerNum - 1] = tmp;
				
				
				
			}
			

		}
		
		answer = players;
        return answer;
    }
}

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js

function solution(players, callings) {
    let playerMap = new Map();
    for (let i = 0; i < players.length; i++) playerMap.set(players[i], i);
    for (let call of callings){
        let rightIndex = playerMap.get(call);
        let leftIndex = playerMap.get(players[rightIndex - 1]);
        playerMap.set(call, leftIndex);
        playerMap.set(players[rightIndex - 1], rightIndex);
        let tmp = players[rightIndex];
        players[rightIndex] = players[leftIndex];
        players[leftIndex] = tmp;
    }
    return players;
}